libc/test/include/IsFiniteTest.h

```diff
@@ -0,0 +1,59 @@
+//===-- Utility class to test the isfinite macro [f|l] ----------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_LIBC_TEST_INCLUDE_MATH_ISFINITE_H
+#define LLVM_LIBC_TEST_INCLUDE_MATH_ISFINITE_H
+
+#include "test/UnitTest/FEnvSafeTest.h"
+#include "test/UnitTest/FPMatcher.h"
+#include "test/UnitTest/Test.h"
+
+#include "include/llvm-libc-macros/math-function-macros.h"
+
+#define PI 3.14159265358979323846
```

Comment:
Only using pi here as an example but really any double precision floating point would suffice for testing. Is including `math.h` or `,<cmath>` acceptable for this or a hex constant in a common header (or for each file it's used, not sure of best practice in this case)

Reply:
This should ideally be defined as `static constexpr T pi = ...` inside the class.

Reply:
Also don't use `PI` because the implicit type conversions in the tests for various floating point types will trigger warnings.  If you want some non zero constants, you can use `T(1)`, `T(-1)`, or `min/max_normal`, `min/max_denormal`.